src/v/cloud_storage/partition_manifest.cc

```diff
@@ -243,8 +292,68 @@ bool partition_manifest::contains(const segment_name& name) const {
     return _segments.contains(key);
 }
 
+void partition_manifest::delete_replaced_segments() { _replaced.clear(); }
+
+bool partition_manifest::advance_start_offset(model::offset new_start_offset) {
+    if (new_start_offset > _start_offset && !_segments.empty()) {
+        key k = {
+          .base_offset = new_start_offset,
+          .term = model::term_id::max(),
+        };
+        auto it = _segments.upper_bound(k);
+        if (it == _segments.begin()) {
+            return false;
+        }
+        it = std::prev(it);
+        if (it->second.committed_offset < new_start_offset) {
```

Comment:
Question: In this code path, is `new_start_offset` in a gap in manifest? 

If so, would placing `_start_offset` at `new_start_offset` would result in it being inside the gap and misaligned with any segments in the manifest? 

Should it be placed at the end of the gap, i.e. `std::next(it)->first.base_offset` (with the appropriate checks)?

Reply:
fixed that, also added a test